### PR TITLE
Fix pytest.warns(None) is deprecated warning

### PR DESCRIPTION
**Pull request by hakancelikdev**

### CODE REVIEW

File: tests/test_warning.py [hunk 1]
```diff
@@ -7,11 +8,19 @@
...
+def test_import_warning(optimization_level, expected_warnings):
+    with pytest.warns() as record:
         with patch("sys.flags", SimpleNamespace(optimize=optimization_level)):
             _check_asserts()
 
     assert len(record) == expected_warnings
+
+
+@pytest.mark.parametrize("optimization_level, expected_warnings", [(0, 0)])
+def test_import_no_warning(optimization_level, expected_warnings):
```
Comment (isidentical) [inline tests/test_warning.py:22]:
```suggestion
@pytest.mark.parametrize("optimization_level", [0])
def test_import_no_warning(optimization_level):
```

**Review by isidentical — APPROVED** [review]

File: tests/test_warning.py (hunk 1 above)
Comment (isidentical) [reply tests/test_warning.py:22]: We don't need the `expected_warnings`, since it is always going to be 0 (and we don't use it in the code).

**Comment by isidentical** [comment]
Thanks for the PR @hakancelikdev, looks good. One nit above ^

**Comment by hakancelikdev** [comment]
@isidentical thank you for review, I fixed

**Pull request merged**